server/public/model/config.go

```diff
@@ -3749,8 +3749,8 @@ func (s *SamlSettings) isValid() *AppError {
 			return NewAppError("Config.IsValid", "model.config.is_valid.saml_idp_url.app_error", nil, "", http.StatusBadRequest)
 		}
 
-		if *s.IdpDescriptorURL == "" || !IsValidHTTPURL(*s.IdpDescriptorURL) {
-			return NewAppError("Config.IsValid", "model.config.is_valid.saml_idp_descriptor_url.app_error", nil, "", http.StatusBadRequest)
+		if err := IsValidURI(*s.IdpDescriptorURL); err != nil {
```

Comment:
What happens if `ldapDescriptorURL` is `nil`?

Reply:
Well this could happen by programmaticall running something like:

```
cfg := &SamlSettings{}
cfg.IsValid()
```

But in the case of mattermost, it's never happening because `SamlSettings.SetDefault()` is always called before and set this to an empty string if it does not exist.

The whole config validation (not just SamlSettings) relies on the fact that `SetDefault` will be called before.

Reply:
Thanks for the clarification. :+1: For some reason, you removed an `== nil` check, which you didn't do.